packages/code-studio/src/dashboard/linker/Linker.tsx

```diff
@@ -29,15 +29,85 @@ import {
 } from '../events';
 import LayoutUtils from '../../layout/LayoutUtils';
 import LinkerOverlayContent from './LinkerOverlayContent';
-import LinkerUtils from './LinkerUtils';
-import LinkType from './LinkType';
+import LinkerUtils, { Link, LinkColumn, LinkType } from './LinkerUtils';
 import { PanelManager } from '../panels';
-import { UIPropTypes } from '../../include/prop-types';
 
 const log = Log.module('Linker');
 
-class Linker extends Component {
-  constructor(props) {
+export type PanelProps = {
+  glContainer: GoldenLayout.Container;
+  glEventHub: GoldenLayout.EventEmitter;
+};
+
+export type Panel = Component<PanelProps>;
+
+export type LinkFilterMapValue = {
+  columnType: string;
+  text: string;
+  value: unknown;
+};
+
+export type LinkFilterMap = Map<string, LinkFilterMapValue>;
+
+export type LinkDataMapValue = {
+  type: string;
+  text: string;
+  value: string;
+};
+
+export type LinkDataMap = Record<string, LinkDataMapValue>;
+
+export type LinkablePanel = Panel & {
+  setFilterMap: (filterMap: LinkFilterMap) => void;
+  unsetFilterValue: () => void;
```

Comment:
Yea true. I think I prefer the arrow style, since it's the similar syntax as if you're doing any other type (can mark readonly, make it optional, etc), and I think you'd have to do it like that anyway if you were doing a union or anything like that (such as `ContextAction[] | (() => ContextAction[])`. 
I don't really know the advantages the other way - you save a couple chars, and when you hover over it, it shows up as `method` instead of `property`... there's probably something I'm missing.

Reply:
https://stackoverflow.com/questions/39156315/function-property-vs-method

Looks like you can apply readonly to a property, but not a method. Also there are some more strict checks if generics are used for a property vs if it is a method.

Linked in that thread is this page which has an example https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#example

So it looks like as a property might be the slightly safer type

Reply:
I'll stick with property for now